docker/build_artifacts/sagemaker/python_service.py

```diff
@@ -95,172 +277,89 @@ def handler(data, context):
-        except FileNotFoundError:
+        if model_name not in self._model_tfs_pid:
             res.status = falcon.HTTP_404
-            res.body = '{} not loaded yet.'.format(model_name)
-        except MultiModelException as multi_model_exception:
-            if multi_model_exception.code == 408:
-                res.status = falcon.HTTP_408
-                res.body = multi_model_exception.msg
-            else:
-                raise MultiModelException(falcon.HTTP_500, multi_model_exception.msg)
-
-    def _read_model_config(self):
-        models = []
-        name_key = re.compile(r'([ \t]*)name:(.*)')
-        uri_key = re.compile(r'([ \t]*)base_path:(.*)')
-        pattern = r'"([A-Za-z0-9_\./\\-]*)"'
-        with open(MODEL_CONFIG_FILE_PATH, 'r') as f:
-            line = f.readline()
-            while line:
-                if name_key.search(line):
-                    model_name = re.search(pattern, line).group().strip('\"')
-                    line = f.readline()
-                    if uri_key.search(line):
-                        uri = re.search(pattern, line).group().strip('\"')
-                        models.append(json.dumps({
-                            'modelName': model_name,
-                            'modelUrl': uri
-                        }))
-                    else:
-                        raise ValueError('Malformed model-config.cfg file.')
-                line = f.readline()
-        return models
+            res.body = json.dumps({
+                'error': 'Model {} is not loaded yet'.format(model_name)
+            })
+        else:
+            try:
+                self._model_tfs_pid[model_name].kill()
+                os.remove('/sagemaker/tfs-config/{}/model-config.cfg'.format(model_name))
+                os.rmdir('/sagemaker/tfs-config/{}'.format(model_name))
+                release_rest_port = self._model_tfs_rest_port[model_name]
+                release_grpc_port = self._model_tfs_grpc_port[model_name]
+                with lock():
+                    bisect.insort(self._tfs_ports['rest_port'], release_rest_port)
+                    bisect.insort(self._tfs_ports['grpc_port'], release_grpc_port)
+                del self._model_tfs_rest_port[model_name]
+                del self._model_tfs_grpc_port[model_name]
+                del self._model_tfs_pid[model_name]
+                res.status = falcon.HTTP_200
+                res.body = json.dumps({
+                    'success': 'Successfully unloaded model {}.'.format(model_name)
+                })
+            except OSError as error:
+                res.status = falcon.HTTP_500
+                res.body = json.dumps({
+                    'error': str(error)
+                }).encode('utf-8')
 
     def validate_model_dir(self, model_path):
         # model base path doesn't exits
         if not os.path.exists(model_path):
             return False
         # model versions doesn't exist
+        log.info("model path exists")
```

Comment:
debug level logging? possibly applies to other info level logging, too

Reply:
This should be removed. I used it for debugging in development.